backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/UserService.cs

```diff
@@ -59,90 +59,121 @@ public async Task<UserDto> AddTagsAsync(List<TagDto> tags)
+    {
+        var users = _context.Users
+            .Include(u => u.Tags)
+            .Include(u => u.Image);
+        var filter = _mapper.Map<UserFilter>(userFilter);
+
+        IQueryable<User> filteredUsers = users;
+
+        if (filter.Language is not null)
+        {
+            filteredUsers = filteredUsers.Where(u => u.Language == filter.Language);
+        }
+        if (filter.LangLevels is not null && filter.LangLevels.Any())
+        {
+            filteredUsers = filteredUsers.Where(u => filter.LangLevels.Contains(u.LanguageLevel));
+        }
+        if (filter.Topics is not null && filter.Topics.Any())
+        {
+            filteredUsers = filteredUsers.Where(u => u.Tags.Any(t => filter.Topics.Contains(t.Name)));
+        }
+        var filteredUsersList = await filteredUsers.ToListAsync();
+        return _mapper.Map<List<UserShortInfoDto>>(filteredUsersList);
+    }
+
+    public async Task<LessonDto> EnrollUserToLesson(long lessonId)
+    {
+        var userId = _authService.UserId;
+
+        var user = _context.Users.SingleOrDefault(u => u.Id == userId) ?? throw new ArgumentException($"Failed to find the user with id {userId}");
+        var lesson = _context.Lessons.SingleOrDefault(l => l.Id == lessonId) ?? throw new ArgumentException($"Failed to find the lesson with id {lessonId}");
+
+        user.Lessons.Add(lesson);
+
+        await _context.SaveChangesAsync();
+
+        void AfterMapAction(Lesson o, LessonDto dto) => dto.SubscribersCount = _context.Lessons
+            .Select(t => new { Id = t.Id, SbCount = t.Subscribers.Count })
+            .FirstOrDefault(l => l.Id == lessonId)!.SbCount;
+
+        return _mapper.Map<Lesson, LessonDto>(lesson, options => options.AfterMap(AfterMapAction));
+    }
+
+    public async Task<string> UploadProfilePhoto(IFormFile file)
+    {
+        var user = await _context.Users.FirstOrDefaultAsync(u => u.Id == _authService.UserId);
+
+        if (user == null)
+        {
+            throw new ArgumentNullException("This user not found");
+        }
+
+        var fileDto = new NewEasySpeakFileDto()
+        {
+            Stream = file.OpenReadStream(),
+            FileName = file.FileName
+        };
+
+        var uploadFileDto = await _fileService.AddFileAsync(fileDto);
+        var profilePhoto = await _context.EasySpeakFiles.FirstOrDefaultAsync(f => f.Id == uploadFileDto.Id);
+
+        if (profilePhoto == null || profilePhoto.Url == null)
+        {
+            throw new ArgumentNullException("This file not found");
+        }
+
+        user.ImageId = uploadFileDto.Id;
+        profilePhoto.UserId = user.Id;
+        await _context.SaveChangesAsync();
+
+        return profilePhoto.Url;
+    }
+
+    private async Task<string> GetProfileImageUrl(long? imageId)
+    {
+        var profileImage = await _context.EasySpeakFiles.FirstOrDefaultAsync(f => f.Id == imageId);
+        return profileImage?.Url ?? "";
+    }
```

Comment:
No fix, just a question
Why github decided that it's your changes?

Reply:
This is a screen from `development`branch. It seems some guys can't use Visual studio formatting commands 
![image](https://user-images.githubusercontent.com/122675547/229278129-f8ea144b-af76-454d-8a56-015b84709344.png)
if you checkout to development branch and 
![image](https://user-images.githubusercontent.com/122675547/229278208-27d5b654-f2de-493e-86ff-a784e4f7f5e7.png)
you'll see this magic)). I've struggled a lot while merging..

Reply:
Ok, interesting
We will need to check formatting of others, thanks)